python/dgl/nn/pytorch/conv/cugraph_relgraphconv.py

```diff
@@ -0,0 +1,388 @@
+"""Torch Module for Relational graph convolution layer using the
+aggregation primitives in cugraph-ops"""
+# pylint: disable=no-member, arguments-differ, invalid-name, too-many-arguments
+import math
+
+import torch as th
+from torch import nn
+
+try:
+    from pylibcugraphops.legacy.aggregators.node_level import (
+        agg_hg_basis_post_bwd_int32,
+        agg_hg_basis_post_bwd_int64,
+        agg_hg_basis_post_fwd_int32,
+        agg_hg_basis_post_fwd_int64,
+    )
+    from pylibcugraphops.legacy.structure.graph_types import (
+        message_flow_graph_hg_csr_int32,
+        message_flow_graph_hg_csr_int64,
+    )
+except ModuleNotFoundError:
+    raise ModuleNotFoundError(
+        "dgl.nn.CuGraphRelGraphConv requires pylibcugraphops to be installed."
+    )
+
+CUDA_SM_PER_BLOCK = 49152
+
+
+class RelGraphConvAgg(th.autograd.Function):
+    r"""Custom autograd function for R-GCN aggregation layer that uses the
+    aggregation functions in cugraph-ops."""
+
+    @staticmethod
+    def forward(ctx, g, fanout, num_rels, edge_types, feat, coeff):
```

Comment:
What is `ctx`? Why did you not document it in the `Parameters` section below?

Reply:
ctx is a context object that can be used to stash information for backward computation in pytorch.
I did not add it to docstring since ctx is not something that has to be passed in by users.

Reply:
This class is not exposed to users. Meanwhile, it's still better to have a brief description for future developers.

Reply:
Fixed.